src/ImageSharp/Common/Extensions/Vector4Extensions.cs

```diff
@@ -12,6 +12,32 @@ namespace SixLabors.ImageSharp
     /// </summary>
     internal static class Vector4Extensions
     {
+        /// <summary>
+        /// Premultiplies the "x", "y", "z" components of a vector by its "w" component leaving the "w" component intact.
+        /// </summary>
+        /// <param name="source">The <see cref="Vector4"/> to premultiply</param>
+        /// <returns>The <see cref="Vector4"/></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector4 Premultiply(this Vector4 source)
+        {
+            float w = source.W;
+            Vector4 premultiplied = source * w;
+            return new Vector4(premultiplied.X, premultiplied.Y, premultiplied.Z, w);
```

Comment:
We can drop an unnecessary constructor invocation for perf here:
```C#
premultiplied.W = w;
return premultiplied;
```

Reply:
I was actually wondering about that when I wrote it. Should have profiled but I didn't know whether requesting the property twice was slower.